test/unit/operations/map_reduce.test.ts

```diff
@@ -0,0 +1,80 @@
+import { expect } from 'chai';
+
+import { MongoClient } from '../../../src';
+import { isHello } from '../../../src/utils';
+import { cleanup, createServer, HELLO } from '../../tools/mongodb-mock';
+
+describe('MapReduceOperation', () => {
+  context('bypass document validation', function () {
+    const test = {
+      server: null
+    };
+    beforeEach(() =>
+      createServer().then(server => {
+        test.server = server;
+      })
+    );
+    afterEach(() => cleanup());
+
+    function testMapReduce(config, done) {
+      const client = new MongoClient(`mongodb://${test.server.uri()}/test`);
+      let close = e => {
+        close = () => null;
+        client.close(() => done(e));
+      };
+
+      test.server.setMessageHandler(request => {
+        const doc = request.document;
+        if (doc.mapReduce) {
+          try {
+            expect(doc.bypassDocumentValidation).equal(config.expected);
+            request.reply({
+              results: 't',
+              ok: 1
+            });
+          } catch (e) {
+            close(e);
+          }
+        }
+
+        if (isHello(doc)) {
+          request.reply(Object.assign({}, HELLO));
+        } else if (doc.endSessions) {
+          request.reply({ ok: 1 });
+        }
+      });
+
+      client.connect(function (err, client) {
+        expect(err).to.not.exist;
+        const db = client.db('test');
+        const collection = db.collection('test_c');
+
+        const options = {
+          out: 'test_c',
+          bypassDocumentValidation: config.actual
+        };
+
+        collection.mapReduce(
```

Comment:
Since we are calling the `mapReduce` method on the `collection` and not directly instantiating the operation in this test, I'd move these tests to `test/unit/collection.test.js`